### PR TITLE
feat: auction event notifications

### PR DESCRIPTION
**Pull request by hkarani**
**PR Summary**

This PR expands seller-side auction notifications and refreshes the auction list item UI for clarity and consistency.

**What Changed**

1. Added end-to-end tracking for auction notification streams:

- Auction goes live (scheduled to live transition)
- Auction ended / settlement begins
- Auction bids
- Auction live chat messages
- Auction thread comments (kind 1111 tied to auction context)
- Product thread comments (kind 1111 tied to product context)

2. Expanded notification state management in [[notifications.ts](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html):

- Added dedicated unseen counters for each notification family
- Added separate last-seen timestamps and mark-seen actions
- Updated recalculation logic so counts remain correct after refresh/reload

3. Updated monitoring and event ingestion in [[useNotificationMonitor.ts](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html):

- Added handlers/subscriptions for live chat comments and both comment target types
- Added lifecycle notification handling for auction live and settlement-begins states
- Removed noisy live-chat toast behavior
- Kept polling fallback aligned with realtime subscriptions

4. Updated dashboard badge aggregation:

- [[_dashboard-layout.tsx](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)
- [[Header.tsx](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)

5. Added seen-state clearing on relevant pages:

- Auctions page clears auction-related notification buckets in [[auctions.tsx](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)
- Products page clears product comment notifications in [[products.tsx](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)

6. Auction list item UI polish in [[auctions.tsx](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html):

- Styled list items to match dashboard auction detail visual language
- Moved status badge to the right-side header area (outside activity panel)
- Added visible outer border treatment
- Restyled top-bid area:
- Improved no-bids empty state
- Green winning-bid treatment with badge
- Bidder avatar made clickable to profile
- Removed non-functional scheduled Manage button

7. Small query support update in [[auctions.tsx](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) to support the new notification flow.

### CODE REVIEW

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 1]
```diff
@@ -40,33 +43,94 @@ import {
...
 	return 'Live'
 }
 
 function formatMaybeDate(timestamp: number): string {
 	if (!timestamp) return 'N/A'
-	return new Date(timestamp * 1000).toLocaleString()
+	return new Date(timestamp * 1000).toLocaleDateString('en-US', { day: 'numeric', month: 'long', year: 'numeric' })
+}
+
+function formatTimeAgo(timestamp: number): string {
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:60]: Let's move this function, alongside `formatMaybeDate` into a proper location, such as lib/utils/format.ts

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 2]
```diff
@@ -40,33 +43,94 @@ import {
...
+
+	if (!topBid) {
+		return (
+			<div className="rounded-xl border border-dashed border-zinc-300 bg-zinc-50 px-5 py-4">
+				<p className="text-sm text-zinc-500">No bids yet. Winning bid will appear here.</p>
+			</div>
+		)
+	}
+
+	return (
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:110]: Why are we redefining another bid display component? Let's make the existing one accessible and reuse it.

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 3]
```diff
@@ -40,33 +43,94 @@ import {
...
+					<span>Top bid:</span>
+					<span>{formatTimeAgo(topBid.created_at ?? 0)}</span>
+				</div>
+				<p className="mt-1 text-2xl font-semibold tracking-tight text-zinc-950">{getBidAmount(topBid).toLocaleString()} sats</p>
+			</div>
+			<div className="mt-2 flex items-center justify-between gap-2">
+				<Badge className="border-emerald-300 bg-white text-emerald-800 hover:bg-white">Winning bid</Badge>
+
+				<div className="flex items-center gap-2">
+					<Link to="/profile/$profileId" params={{ profileId: topBid.pubkey }}>
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:123]: Use <UserCard> instead of redefining another avatar + name component. (This wouldn't apply if we re-use bid component)

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx (hunk 2 above)
Comment (Franchovy) [reply src/routes/_dashboard-layout/dashboard/products/auctions.tsx:110]: Needs to be extracted from src/components/AuctionBidsContainer.tsx.

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 4]
```diff
@@ -77,196 +141,99 @@ function AuctionListBody({
...
+			<div className="flex flex-col gap-6 lg:flex-row">
+				<div className="flex shrink-0 items-start gap-3 lg:flex-col">
+					{thumbnailUrl ? (
+						<img src={thumbnailUrl} alt="" className="h-24 w-24 shrink-0 rounded-xl object-cover lg:h-32 lg:w-32" />
+					) : (
+						<div className="flex h-24 w-24 shrink-0 items-center justify-center rounded-xl bg-zinc-100 lg:h-32 lg:w-32">
+							<Gavel className="h-8 w-8 text-zinc-400" />
+						</div>
+					)}
+					<Link to="/dashboard/products/auctions/$auctionId" params={{ auctionId: auction.id }} className="lg:w-32">
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:186]: Let's link directly to the public view.

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 5]
```diff
@@ -77,196 +141,99 @@ function AuctionListBody({
...
-						)}
-					>
+
+				<div className="flex w-full shrink-0 flex-col items-end gap-3 lg:w-64">
+					<span className={cn('inline-flex w-fit rounded-full border px-3 py-1 text-xs font-medium', STATUS_BADGE_STYLES[status])}>
 						{status}
 					</span>
+					<div className="w-full space-y-3 rounded-xl border border-zinc-200 bg-zinc-50 px-5 py-5">
+						<p className="text-[11px] font-semibold uppercase tracking-[0.18em] text-muted-foreground">Activity</p>
+						<ActivityRow header="Bids" unit="Bid" count={bidsCount} newCount={newBidsCount} />
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:229]: The dashboard auctions view automatically clears all new notifications, marking them as "seen". This makes it impossible to see the "newCount" on any of the specific auctions and their activities.

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 6]
```diff
@@ -77,196 +141,99 @@ function AuctionListBody({
...
 
-				<div className="flex flex-wrap items-center gap-2 pt-1">
-					<Link to="/dashboard/products/auctions/$auctionId" params={{ auctionId: auction.id }}>
-						<Button variant="outline" size="sm" className="gap-2">
-							<Eye className="w-3.5 h-3.5" />
-							View Auction
+	return (
+		<div className="rounded-lg border border-zinc-200 bg-background p-6 shadow-md">
+			<div className="flex flex-col gap-6 lg:flex-row">
+				<div className="flex shrink-0 items-start gap-3 lg:flex-col">
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:178]: The auction image is slightly too small, the problem being the button underneath being squashed. See if you can expand the width a little bit to give it space to breathe.

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 7]
```diff
@@ -77,196 +141,99 @@ function AuctionListBody({
...
-	const biddingCutoffAt = getAuctionBiddingCutoffAt(auction)
-	const now = Math.floor(Date.now() / 1000)
-	const status = formatAuctionStatus(startAt, biddingCutoffAt, now)
-	const images = getAuctionImages(auction)
-	const thumbnailUrl = images.length > 0 ? images[0][1] : null
+				<div className="flex-1 min-w-0 space-y-4">
+					<div className="min-w-0">
+						<h3 className="text-lg font-semibold text-foreground truncate">{getAuctionTitle(auction)}</h3>
+						<p className="text-sm text-muted-foreground truncate">{summary}</p>
+						<p className="mt-1 text-xs text-muted-foreground">Created: {formatMaybeDate(auction.created_at ?? 0)}</p>
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:198]: `formatMaybeDate` doesn't include the time, which may be useful in this context.

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 8]
```diff
@@ -77,196 +141,99 @@ function AuctionListBody({
...
+					</div>
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:199]: Would be great to also have the Auction Progress widget here (I know, it wasn't part of the original spec... but it feels like it's missing)

File: src/lib/stores/notifications.ts [hunk 9]
```diff
@@ -13,6 +13,11 @@ export interface NotificationState {
 	unseenMessages: number // New messages in conversations
 	unseenPurchases: number // Updates to orders where user is buyer
 	unseenAuctionBids: number // New bids on auctions where user is seller
+	unseenAuctionComments: number // New live-chat comments on seller auctions
```
Comment (Franchovy) [inline src/lib/stores/notifications.ts:16]: Can we rename these to be clearer and avoid potential confusion?

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 10]
```diff
@@ -293,6 +258,10 @@ function AuctionsOverviewComponent() {
 	useEffect(() => {
 		if (isAuthenticated && user?.pubkey) {
 			notificationActions.markAuctionBidsSeen()
+			notificationActions.markAuctionCommentsSeen()
+			notificationActions.markAuctionEventCommentsSeen()
+			notificationActions.markAuctionLiveSeen()
+			notificationActions.markAuctionSettlementBeginsSeen()
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:264]: We should reconsider whether it makes sense to clear all notifications here when the user doesn't have time to dig into any one notification's source.

File: src/routes/_dashboard-layout.tsx [hunk 11]
```diff
@@ -170,7 +184,18 @@ function DashboardLayout() {
...
+		unseenMessages,
+		unseenPurchases,
+		unseenAuctionBids,
+		unseenAuctionComments: unseenAuctionLiveChatComments,
+		unseenAuctionEventComments: unseenAuctionThreadComments,
+		unseenProductComments: unseenProductThreadComments,
+		unseenAuctionLive: unseenScheduledAuctionsNowLive,
+		unseenAuctionSettlementBegins,
+		unseenBidUpdates,
+	} = useStore(notificationStore)
```
Comment (Franchovy) [inline src/routes/_dashboard-layout.tsx:198]: `useStore` is deprecated. Use `useSelector` instead.

File: src/hooks/useNotificationMonitor.ts [hunk 12]
```diff
@@ -44,10 +53,65 @@ const getAuctionCoordinate = (auction: NDKEvent): string => {
 	return auctionDTag ? `30408:${auction.pubkey}:${auctionDTag}` : ''
 }
 
+const getProductCoordinate = (product: NDKEvent): string => {
```
Comment (Franchovy) [inline src/hooks/useNotificationMonitor.ts:56]: This is a redefinition of an existing function. You can find it as `getProductCoordinates` in src/queries/products.tsx

File: src/hooks/useNotificationMonitor.ts [hunk 13]
```diff
@@ -44,10 +53,65 @@ const getAuctionCoordinate = (auction: NDKEvent): string => {
...
+	const startAt = getAuctionStartAt(auction)
+	const createdAt = auction.created_at ?? 0
+	return startAt > 0 && startAt > createdAt
+}
+
+const collectSellerAuctionLifecycleNotifications = (
+	sellerAuctions: NDKEvent[],
+	lastSeenAuctionLive: number,
+	lastSeenAuctionSettlementBegins: number,
+	seenAuctionLiveEventIds: Set<string>,
```
Comment (Franchovy) [inline src/hooks/useNotificationMonitor.ts:75]: I would recommend renaming "liveEvents" here since it seems to refer to the live event + live chat. You can rename it auctionStart or auctionStartScheduled or similar.

File: src/hooks/useNotificationMonitor.ts [hunk 14]
```diff
@@ -112,6 +176,94 @@ const fetchTaggedAuctionEvents = async ({
...
+	since,
+}: {
+	productCoordinates: string[]
+	productEventIds: string[]
+	since?: number
+}): Promise<NDKEvent[]> => {
+	const filters: NDKFilter[] = []
+	const baseFilter = {
+		kinds: [COMMENT_KIND_NDK],
+		limit: AUCTION_MONITOR_LIMIT,
```
Comment (Franchovy) [inline src/hooks/useNotificationMonitor.ts:191]: Why is AUCTION_MONITOR_LIMIT implemented in product related function? Either rename the var or change the reference.

File: src/hooks/useNotificationMonitor.ts [hunk 15]
```diff
@@ -112,6 +176,94 @@ const fetchTaggedAuctionEvents = async ({
...
+		...(typeof since === 'number' ? { since } : {}),
+	}
+
+	for (const coordinateChunk of chunkValues(productCoordinates, AUCTION_FILTER_CHUNK_SIZE)) {
```
Comment (Franchovy) [inline src/hooks/useNotificationMonitor.ts:195]: Same as above

File: src/hooks/useNotificationMonitor.ts [hunk 16]
```diff
@@ -112,6 +176,94 @@ const fetchTaggedAuctionEvents = async ({
...
+			...baseFilter,
+			'#a': coordinateChunk,
+		})
+		filters.push({
+			...baseFilter,
+			'#A': coordinateChunk,
+		})
+	}
+
+	for (const eventIdChunk of chunkValues(productEventIds, AUCTION_FILTER_CHUNK_SIZE)) {
```
Comment (Franchovy) [inline src/hooks/useNotificationMonitor.ts:206]: same as above

File: src/hooks/useNotificationMonitor.ts [hunk 17]
```diff
@@ -246,31 +413,85 @@ export const useNotificationMonitor = () => {
...
 					const isUpdate =
 						typeTag?.[1] === ORDER_MESSAGE_TYPE.PAYMENT_REQUEST ||
 						typeTag?.[1] === ORDER_MESSAGE_TYPE.STATUS_UPDATE ||
 						typeTag?.[1] === ORDER_MESSAGE_TYPE.SHIPPING_UPDATE
 
-					// Only count if it's an update type AND it's from the seller (not the buyer)
 					return isUpdate && event.pubkey !== user.pubkey && isNewPurchaseUpdate(event)
 				})
 
 				const sellerAuctions = await fetchAuctionsByPubkey(user.pubkey, 500)
```
Comment (Franchovy) [inline src/hooks/useNotificationMonitor.ts:433]: This function relies on a sequential loading of many different events and event types.

We can improve this by doing a parallel fetch (Promise.all) where possible. Any inner dependency chains can also be parallelised (e.g. one initial fetch -> multiple fetches in parallel)

File: src/hooks/useNotificationMonitor.ts [hunk 18]
```diff
@@ -417,33 +743,110 @@ export const useNotificationMonitor = () => {
 				})
 
 				console.log('[NotificationMonitor] Subscriptions active:', subscriptionsRef.current.length)
+
+				auctionLifecyclePollId = window.setInterval(async () => {
+					try {
+						const pollSellerAuctions = await fetchAuctionsByPubkey(user.pubkey, 500)
```
Comment (Franchovy) [inline src/hooks/useNotificationMonitor.ts:749]: These inner variables are referred to as poll(...), yet they do not poll, they simply fetch, and they are not functions but are the results of the functions. I suggest renaming the variables here appropriately to more clearly reflect the meaning.

**Review by Franchovy — CHANGES_REQUESTED** [review]
This PR successfully expands the notifications surface to cover auctions (as well as product comments). The updated dashboard view for auctions is also appreciated, highlighting the current auction state and settlement actions. The comments made on this PR / changes requested are regarding the current implementation and not aiming to expand the scope.

However it is also important to highlight next steps that should be made:

- Settlement status and actions need to be centralized under one helper that can be reused. Currently the cases of a cancelled / no bids auction settlement, and the griefed/bid fallback scenario are not covered by the existing functionality.
- The dashboard auctions view needs to draw attention to any pending settlement actions at the top.
- The dashboard bid view needs updating to show new/important information as well as settlement actions
- The current notifications system is highly verbose and repetitive. Currently we have about 5 different places where boilerplate code needs to be added to expand the notifications surface, when adding a new notification type should be as trivial as adding a single line that defines a nostr event query and allows the notification count + actual underlying event to be referenced throughout the codebase.
- There is no (e2e, unit) test coverage for notifications.
- There are no jsx comments anywhere in the notifications functionality.
- Notifications have no proper way of routing to the actual event and are instantly cleared on opening the dashboard page. This is a known limitation to the current notifications system and needs addressing.

File: src/routes/_dashboard-layout/dashboard/products/auctions.tsx [hunk 19]
```diff
@@ -40,33 +43,94 @@ import {
...
-/**
- * Compact body for a seller's auction row. The auction detail route is the
- * one place to look at the full thing; this collapsible deliberately keeps
- * only the "what's the headline state" essentials plus the inline settlement
- * action so a seller can close ended auctions without a navigation hop.
- * Anything that resembles tech-debt (event IDs, mint URLs, settlement-policy
- * strings, etc.) lives on the detail route behind accordions.
- */
-function AuctionListBody({
+const STATUS_BADGE_STYLES: Record<AuctionStatus, string> = {
```
Comment (Franchovy) [inline src/routes/_dashboard-layout/dashboard/products/auctions.tsx:77]: These seem like redefinitions of existing styles. Let's move an existing definition into a more accessible place such as lib/auctions/styles and re-use the styleset from there.

```
Based on my search, I can see there are indeed multiple instances of similar badge style definitions in the codebase:

STATUS_BADGE_STYLES in src/routes/_dashboard-layout/dashboard/products/auctions.tsx (the one you added):


const STATUS_BADGE_STYLES: Record<AuctionStatus, string> = {
  Scheduled: 'border-sky-400 text-sky-700 bg-sky-50',
  Live: 'border-emerald-400 text-emerald-700 bg-emerald-50',
  Settlement: 'border-sky-400 text-sky-700 bg-sky-50',
  Ended: 'border-zinc-300 text-zinc-500 bg-zinc-50',
}
AUCTION_STATUS_STYLES in src/routes/_dashboard-layout/dashboard/products/auctions.$auctionId.tsx:

const AUCTION_STATUS_STYLES: Record<string, string> = {
  Live: 'border-emerald-200 bg-emerald-50 text-emerald-800',
  Scheduled: 'border-sky-200 bg-sky-50 text-sky-800',
  Ended: 'border-zinc-200 bg-zinc-100 text-zinc-700',
}
SETTLEMENT_STATUS_STYLES in src/routes/_dashboard-layout/dashboard/products/auctions.$auctionId.tsx:


const SETTLEMENT_STATUS_STYLES: Record<string, string> = {
  settled: 'border-emerald-200 bg-emerald-50 text-emerald-800',
  reserve_not_met: 'border-amber-200 bg-amber-50 text-amber-800',
  cancelled: 'border-rose-200 bg-rose-50 text-rose-800',
  unknown: 'border-zinc-200 bg-zinc-100 text-zinc-700',
}
```

**Review by Franchovy — CHANGES_REQUESTED** [review]
Additional comment

**Comment by maximotodev** [comment]
@codex review

**Comment by c03rad0r** [comment]
@hkarani Excellent work on the auction event notifications system! This is a comprehensive and well-thought-out implementation.

### 👍 Strengths
- **Complete coverage** - Live chat, bids, settlement, comments all handled
- **Good state management** - Separate unseen counters and timestamps per notification family
- **UI polish** - Auction list items look much more professional
- **Proper seen-state clearing** - Notifications cleared when visiting relevant pages
- **Non-breaking** - Expands existing functionality without breaking changes

### 🔍 Review Notes

#### 1. Performance Considerations
The implementation adds multiple new subscriptions:
```typescript
// New handlers in useNotificationMonitor.ts
- live chat comments  
- product thread comments
- auction lifecycle notifications
- bid stream updates
```

**Questions**:
- Are all these subscriptions active simultaneously, or do they follow a "subscribe on page visit" pattern?
- Have you tested memory usage with many concurrent auctions?
- Is there any deduplication logic for overlapping event types?

#### 2. Error Handling
I don't see explicit error handling for:
- Failed subscription attempts
- Malformed notification events  
- State corruption in unseen counters

**Suggestion**: Add try/catch blocks around subscription setup and consider adding validation for notification event structure.

#### 3. Testing Coverage
The description mentions "end-to-end tracking" but I don't see specific test files mentioned.

**Question**: Are there E2E tests covering the notification flows, especially for the edge cases like:
- Notifications arriving while page is active
- State persistence across page refreshes
- Multiple notifications arriving simultaneously?

### ✅ Approval Ready
This looks very close to ready! If you can address the performance/testing questions above and perhaps add some basic error handling, I'd be happy to approve this PR.

Great work on pushing the auction functionality forward! The attention to notification state management and UI polish is excellent. 🚀

**Review by maximotodev — CHANGES_REQUESTED** [review]
Thanks for pushing this forward 👍. The UX direction is useful, and I like that seller auction activity is being surfaced more clearly.

Local smoke:
- Targeted Prettier passes for the changed files.
- `bun run test:unit` passes: 433 pass / 0 fail.
- UI smoke: public auction browsing/detail renders, dashboard auction cards render in both zero-activity and non-zero activity states, and seller auction detail opens. The broad dashboard Auctions page also appears to clear the Auctions badge immediately on entry, which matches the mark-seen concern below.

I think this needs a small follow-up before merge, mostly around notification correctness and relay/query pressure:

1. Realtime comment handlers should still honor last-seen state. Initial fetch filters comments against the relevant `lastSeen*` timestamp, but the realtime handlers only dedupe by event id and skip self-authored events. Since the monitor remains mounted while route-level `mark*Seen()` calls can advance the last-seen timestamp, a delayed relay event from before the clear can increment the unseen count again. Please add the same `isNewAuctionComment` / `isNewAuctionEventComment` / `isNewProductComment` checks inside the realtime handlers.

2. The auction overview appears to fetch bids twice: the parent calls `useAuctionBidsForList(...)` for sorting, then each `AuctionListItem` calls `useAuctionBids(...)` again for display/top-bid/new-bid count. Please reuse/pass the already-fetched list data into the row, or otherwise avoid per-row duplicate bid subscriptions/queries.

3. Please confirm or adjust the mark-seen semantics for `/dashboard/products/auctions`. Right now visiting the broad auctions page clears all auction buckets — bids, live chat, thread comments, scheduled-now-live, and settlement-begins — even though those are per-auction activities. If that is the intended “auction inbox viewed” behavior, a short code comment would help; otherwise this should clear more narrowly or after the relevant data has loaded.

4. Same question for product comments: visiting the broad products page currently marks product comment notifications seen. For comment/thread notifications, it may be safer to clear only when the relevant product/comment context is opened.

Small cleanup: `toast` is imported in `useNotificationMonitor.ts` but does not appear to be used.

Holding approval on the notification correctness/query-pressure items above.
